packages/@aws-cdk/core/lib/cfn-resource.ts

```diff
@@ -272,8 +272,12 @@ export class CfnResource extends CfnRefElement {
   }
 
   protected get cfnProperties(): { [key: string]: any } {
-    const tags = TagManager.isTaggable(this) ? this.tags.renderTags() : {};
-    return deepMerge(this._cfnProperties || {}, {tags});
+    if (TagManager.isTaggable(this) && this.tags.renderTags() !== undefined) {
+      const tags: {[key: string]: any} = {};
+      tags[this.tags.tagPropertyName] = this.tags.renderTags();
```

Comment:
Maybe we can refactor this so that we rename `renderTags` to `applyTagsProperty` or something like that and it will take care of everything:

```
this.tags.applyTagsProperty(this._cfnProperties);
```

Reply:
What do you think about: 

`this.tags.asCfnProperty()` ?